src/iam/provisionmanager/provisionmanager.cpp

```diff
@@ -0,0 +1,135 @@
+#include "aos/iam/provisionmanager.hpp"
+#include "log.hpp"
+
+#include <iostream>
+
+namespace aos::iam::provisionmanager {
+
+/***********************************************************************************************************************
+ * Public
+ **********************************************************************************************************************/
+
+Error ProvisionManager::Init(ProvisionManagerCallback& callback, certhandler::CertHandlerItf& certHandler)
+{
+    LOG_DBG() << "Init provision manager";
+
+    mCallback    = &callback;
+    mCertHandler = &certHandler;
+
+    return aos::ErrorEnum::eNone;
+}
+
+Error ProvisionManager::StartProvisioning(const String& password)
+{
+    LOG_DBG() << "Start provisioning";
+
+    auto err = mCallback->OnStartProvisioning(password);
+    if (!err.IsNone()) {
+        return AOS_ERROR_WRAP(err);
+    }
+
+    CertTypes certTypes;
+
+    if (err = mCertHandler->GetCertTypes(certTypes); !err.IsNone()) {
+        return AOS_ERROR_WRAP(err);
+    }
+
+    for (const auto& certType : certTypes) {
+        LOG_DBG() << "Clear cert storage: type = " << certType;
+
+        if (err = mCertHandler->Clear(certType); !err.IsNone()) {
+            return AOS_ERROR_WRAP(err);
+        }
+
+        LOG_DBG() << "Set owner: type = " << certType;
+
+        if (err = mCertHandler->SetOwner(certType, password); !err.IsNone()) {
+            return AOS_ERROR_WRAP(err);
+        }
+
+        auto certModuleConfig = mCertHandler->GetModuleConfig(certType);
+        if (!certModuleConfig.mError.IsNone()) {
+            return AOS_ERROR_WRAP(certModuleConfig.mError);
+        }
+
+        if (certModuleConfig.mValue.mIsSelfSigned) {
+            LOG_DBG() << "Create self signed cert: type = " << certType;
+
+            if (err = mCertHandler->CreateSelfSignedCert(certType, password); !err.IsNone()) {
+                return AOS_ERROR_WRAP(err);
+            }
+        }
+    }
+
+    LOG_DBG() << "Start disc encryption";
```

Comment:
Let's remove this log: to do not confuse user in case disk encryption is not needed.